private/PROPOSAL/Sector.cxx

```diff
@@ -62,11 +64,71 @@ Sector::Definition::Definition(const Definition& def)
     , location(def.location)
     , utility_def(def.utility_def)
     , cut_settings(def.cut_settings)
-    , medium_(def.medium_->clone())
-    , geometry_(def.geometry_->clone())
+    , medium_(def.medium_)
+    , geometry_(def.geometry_)
 {
 }
 
+Sector::Definition::Definition(const nlohmann::json& config)
+    : utility_def(config)
+{
+    if(not config.is_object()) throw std::invalid_argument("No json object found.");
+
+    std::array<std::pair<std::string, Sector::ParticleLocation::Enum>, 3> cuts = {
+        std::make_pair("cuts_infront", Sector::ParticleLocation::InfrontDetector),
+        std::make_pair("cuts_inside",Sector::ParticleLocation::InsideDetector),
+        std::make_pair("cuts_behind",Sector::ParticleLocation::BehindDetector),
+    };
+
+    for (const auto& cut : cuts) {
+        if(config.contains(cut.first)) {
+            EnergyCutSettings cut_settings(config.at(cut.first));
+            location = cut.second;
+            break;
```

Comment:
why break here?  and why is nothing done with `cut_settings`. this goes out of scope after the if.

Reply:
Energy Cuts has three possible keywords. The keyword itself contains information about the position. 
Construction of `cut_settings` is specified in 3082146. It's a private class variable which break underscores variable definition convention.